pkg/build/builder/common.go

```diff
@@ -448,7 +448,7 @@ func replaceImagesFromSource(node *parser.Node, imageSources []buildapiv1.ImageS
 		}
 	}
 	names := make(map[string]string)
-	stages, err := imagebuilder.NewStages(node, imagebuilder.NewBuilder(nil))
+	stages, err := imagebuilder.NewStages(node, imagebuilder.NewBuilder(names))
```

Comment:
The current version of imagebuilder uses this map by reference, which will conflict with our use of it below.  Can we use a fresh map here, and on line 491?

Reply:
So for line 491 I see `stages, err := imagebuilder.NewStages(node, imagebuilder.NewBuilder(nil))` so I am interpreting the need to pass a non-nil map into that call as well.... granted that is in a different method, `findReferencedImages`

@nalind  - If you are asking for changes around the map within `replaceImagesFromSource` please clarify

Reply:
OK I see what you did in https://github.com/openshift/builder/pull/131/files ... I presume you mean that.

Reply:
updated @nalind thanks (I see the use map by reference element now :-) ) !!